docs/syntax/js.md

```diff
@@ -77,19 +77,40 @@ Modules that expose an `init` method or an instance constructor which takes an a
 
 Where this reference is passed to an `init` function, the module *may* create multiple instances and return them in an array.  Where passed to a constructor, the module must only create one instance and return it.
 
+Where the reference is to an element that is not itself owned DOM, the init function *may* traverse the subtree looking for elements that are.
 
-## Data attributes
+Where JavaScript exists to enhance elements, and accompanying CSS depends on knowing whether the JavaScript intends to apply (or has applied)that enhancement, the JavaScript *may* add a data attribute of the form `data-{modulename}-js` with no value to the root element of the component when the JavaScript initialises.  For example, o-tabs markup would not contain a `o--if-js` class, because the tabs content should remain visible even if the tabs JavaScript is not running on the page, but if the JavaScript does run, it could apply an `data-o-tabs-js` data attribute to allow the tabs CSS to hide all but the selected tab panel.
```

Comment:
Why not mention that data attributes can also be used for config stuff like it used to?